geode-protobuf/src/main/java/org/apache/geode/internal/protocol/ProtobufLocatorPipeline.java

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.internal.protocol;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import org.apache.geode.annotations.Experimental;
+import org.apache.geode.cache.IncompatibleVersionException;
+import org.apache.geode.distributed.internal.InternalLocator;
+import org.apache.geode.internal.cache.tier.sockets.ClientProtocolPipeline;
+import org.apache.geode.internal.cache.tier.sockets.MessageExecutionContext;
+import org.apache.geode.internal.protocol.protobuf.ProtobufStreamProcessor;
+import org.apache.geode.internal.protocol.protobuf.statistics.ProtobufClientStatistics;
+
+@Experimental
+public final class ProtobufLocatorPipeline implements ClientProtocolPipeline {
+  private final ProtobufClientStatistics statistics;
+  private final InternalLocator locator;
+  private final ProtobufStreamProcessor streamProcessor;
+
+  ProtobufLocatorPipeline(ProtobufStreamProcessor protobufStreamProcessor,
+      ProtobufClientStatistics statistics, InternalLocator locator) {
+    this.streamProcessor = protobufStreamProcessor;
+    this.statistics = statistics;
+    this.locator = locator;
+    this.statistics.clientConnected();
+  }
+
+  @Override
+  public void processMessage(InputStream inputStream, OutputStream outputStream)
```

Comment:
For locator, I would like the interface to encode and decode the message.

Reply:
I had originally had that setup; Udo requested that I refactor it to be the same as the client. The end result does about the same.

Reply:
I prefer consistency over continuing special casing a bad situation